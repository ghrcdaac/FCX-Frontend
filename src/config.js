/*
    Following are the settings for FCX that should be configured per deployment environment.
    
    TODO:
    Don't freak out, the default value are left here only to make it easy to get started working with the code base. They are not guaranteed to work.
    In future, actual values of these variables should be directly read from environment variables and defaults should be removed.
*/

const dataBaseUrl = process.env.DATA_BASE_URL || "https://ghrc-fcx-viz-output.s3-us-west-2.amazonaws.com"
const abiBaseUrl = process.env.ABI_BASE_URL || "https://q6eymnfsd9.execute-api.us-west-2.amazonaws.com"
const flightTrackBaseUrl = process.env.FLIGHT_TRACK_BASE_URL || "https://fcx-czml.s3.amazonaws.com"
const mapboxAccessToken = process.env.MAPBOX_ACCESS_TOKEN || "pk.eyJ1IjoiYWppbmt5YWt1bGthcm5pIiwiYSI6ImNrYzY1NHh3bDA3cXIyenA4dDdpOGlja2UifQ.g4r3pjugLK8yC2mkQmJr4w"
const mapboxStyleId = process.env.MAPBOX_STYLE || "ckc65ettm0ixc1ipexrpklr96"
const mapboxUsername = process.env.MAPBOX_USERNAME || "ajinkyakulkarni"
const mapboxUrl = "https://api.mapbox.com/styles/v1/" + mapboxUsername + "/" + mapboxStyleId + "/tiles/256/{z}/{x}/{y}?access_token=" + mapboxAccessToken
const cesiumDefaultAccessToken =
  process.env.CESIUM_DEFAULT_TOKEN || "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIyNmQyZDgxNS0xNzY4LTQ4MzctOWZkOC02YmZkYjQ1ZGI1MDUiLCJpZCI6ODU1MSwic2NvcGVzIjpbImFzciIsImdjIl0sImlhdCI6MTU5NDM4OTIxMn0.WPkQXxpVgW5AfQdZKB15dMnLxQIjmA_Q2qimnDXONvc"
const supportEmail = process.env.GHRC_SUPPORT_EMAIL || "support-ghrc@earthdata.nasa.gov"

export { dataBaseUrl, abiBaseUrl, flightTrackBaseUrl, mapboxUrl, cesiumDefaultAccessToken, supportEmail }
