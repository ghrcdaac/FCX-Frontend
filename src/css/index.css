@import "../../node_modules/cesium/Build/Cesium/Widgets/widgets.css";

body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

a {
  text-decoration: none;
  color: #337ab7;
  background-color: transparent;
}

.th-wrapper {
  position: sticky;
}

.header {
  position: fixed;
}