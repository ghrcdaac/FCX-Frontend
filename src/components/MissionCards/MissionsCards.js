import { useHistory, withRouter } from "react-router"

import Card from "./Card"
import '../../css/cards.css'

const MissionsCards = ({ missions }) => {

  const history = useHistory()
  const basePath = '/fcx'
  return (
    <div className="mission-card-group">
      {
        missions.map(({ name, description, image, dates, landingPageURL, path }) => (
          <Card
            key={name}
            imageUrl={`missions-logos/${image}`}
            name={name}
            description={description}
            reference={landingPageURL}
            dates={dates}
            onClickHandler={history.push}
            args={[`${basePath}${path}`]}
          />

        ))
      }
    </div>)
}

export default withRouter(MissionsCards)