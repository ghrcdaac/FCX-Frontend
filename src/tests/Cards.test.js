import { shallow } from "enzyme";
import toJSON from "enzyme-to-json"
import React from "react";

import '../customized-components/react-timeline-9000/src/setupTests'

import Card from '../components/MissionCards/Card'
import MissionsCards from '../components/MissionCards/MissionsCards'

describe("Cards Components", () => {
  describe("Card", () => {
    it("Should Match Card Snapshot", () => {
      const wrapper = shallow(<Card />)
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })
  describe("Mission Cards", () => {
    it("Should Match Mission Card Snapshot", () => {
      const wrapper = shallow(<MissionsCards />)
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })
})