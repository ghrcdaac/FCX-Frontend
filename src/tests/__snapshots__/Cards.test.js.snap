// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Components Card Should Match Card Snapshot 1`] = `
<div
  className="mission-card"
>
  <div
    className="mission-card-side"
  >
    <img
      className="mission-card-image"
    />
  </div>
  <div
    className="mission-card-body"
  >
    <h4
      className="mission-card-name"
      onClick={[Function]}
    />
    <div
      className="mission-card-description"
    />
  </div>
</div>
`;

exports[`Cards Components Mission Cards Should Match Mission Card Snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
